### PR TITLE
Always use sets for block values

### PR DESCRIPTION
**Pull request by jpreese**
For https://github.com/tmccombs/hcl2json/issues/18

Disclaimer: 99.9% of the change is just always wrapping `value` on line `125` (https://github.com/tmccombs/hcl2json/blob/master/convert/convert.go#L125) inside of `[]interface{}`.

The other diffs are mostly from me just trying to understand and document the behavior.

Without going into Terraform-specific checks, this feels like the path forward. For Terraform, it might be a little silly for some blocks, like variables:

```hcl
variable "my_var" {
  type = string
}
```

will produce

```json
"variable": {
  "my_var": [
    {
       "type": "${string}"
    }
   ]
}
```

Even though its impossible for it to have more than one. But in HCL, a block is a block.

### CODE REVIEW

File: convert/convert.go [hunk 1]
```diff
@@ -49,80 +49,105 @@ func File(file *hcl.File, options Options) ([]byte, error) {
...
 
 func (c *converter) convertBody(body *hclsyntax.Body) (jsonObj, error) {
-	var err error
 	out := make(jsonObj)
-	for key, value := range body.Attributes {
-		out[key], err = c.convertExpression(value.Expr)
-		if err != nil {
-			return nil, err
+
+	for _, block := range body.Blocks {
```
Comment (tmccombs) [inline convert/convert.go:79]: Does the order of converting bodies and attributes matter?

File: convert/convert.go [hunk 2]
```diff
@@ -49,80 +49,105 @@ func File(file *hcl.File, options Options) ([]byte, error) {
...
+		//
+		// To create the JSON representation, check to see if the label exists in the current output:
+		//
+		// When the label exists, move onto the next label reference.
+		// When a label does not exist, create the label in the output and set that as the next label reference
+		// in order to append (potential) labels to it.
+		if _, exists := out[key]; exists {
 			var ok bool
-			out, ok = inner.(jsonObj)
+			out, ok = out[key].(jsonObj)
```
Comment (tmccombs) [inline convert/convert.go:125]: why did you change this?

File: convert/convert.go [hunk 3]
```diff
@@ -49,80 +49,105 @@ func File(file *hcl.File, options Options) ([]byte, error) {
...
 			if !ok {
-				// TODO: better diagnostics
 				return fmt.Errorf("Unable to convert Block to JSON: %v.%v", block.Type, strings.Join(block.Labels, "."))
 			}
 		} else {
-			obj := make(jsonObj)
-			out[key] = obj
-			out = obj
+			out[key] = make(jsonObj)
+			out = out[key].(jsonObj)
```
Comment (tmccombs) [inline convert/convert.go:131]: again, why change?

**Review by tmccombs — APPROVED** [review]

File: convert/convert.go (hunk 1 above)
Comment (jpreese) [reply convert/convert.go:79]: It does not, no. This must have just gotten moved as I was trying to make sense of the codebase and how the HCL parser bucketed each part.

File: convert/convert.go (hunk 3 above)
Comment (jpreese) [reply convert/convert.go:131]: For both of these, this tripped me up for quite a bit. 

When using `out[key] =`, we're actually changing the value passed in by the user.
When using `out =`, we're just setting the "next" out value to get the next label.

All of this makes sense in hindsight, but creating the single `obj` and setting it to both `out[key]` as well as `out` wasn't immediately clear. 

This feels more natural in that we're creating a new value at `out[key]` and then setting `out` to that value that we _just created_.

I'm more than happy to revert any of this. It really was just trying to organize things in my head, but I don't want to make it confusing for you if the old way made more sense to you.
